phone_proxy/lib/main.dart

```diff
@@ -1,149 +1,231 @@
-    }).onError(print);
+    }).onError((error) { logger.e('error listening for connections: $error'); });
 
     unawaited(socksServerProxy
         .addServerSocket(L2CapChannelServerSocketUtils.multiplex(chan)));
   }).asFuture();
 }
 
-Future<void> manageMachine(BleCentral bleCentral, String machineName) async {
-  print('scanning for $machineName now');
+Future<void> manageMachine(BleCentral bleCentral, String mobileDevice, machineToManage) async {
+  logger.i('scanning for $machineToManage now');
   late StreamSubscription<DiscoveredBlePeripheral> deviceSub;
   deviceSub = bleCentral.scanForPeripherals([viamSvcUUID]).listen(
     (periphInfo) {
       deviceSub.pause();
-      print('found ${periphInfo.name}; connecting...');
+      logger.i('found ${periphInfo.name}; connecting');
       bleCentral.connectToPeripheral(periphInfo.id).then((periph) async {
-        print('connected to $machineName');
+        logger.i('connected to $machineToManage');
 
-        final viamSvc = periph.services.cast<BleService?>().firstWhere(
-            (svc) => svc != null && svc.id == viamSvcUUID,
-            orElse: () => null);
+        BleService? viamSvc;
+        for (int i = 0; i < numRetries; i++) {
+          viamSvc = periph.services.cast<BleService?>().firstWhere(
+              (svc) => svc != null && svc.id == viamSvcUUID,
+              orElse: () => null);
+          if (viamSvc != null) {
+            break;
+          }
+        }
         if (viamSvc == null) {
-          // Note(erd): this could use some retry logic
-          print("viam service missing; disconnecting");
+          logger.e("expected service missing; disconnecting");
           await periph.disconnect();
           deviceSub.resume();
           return;
         }
 
-        final periphNameChar = viamSvc.characteristics
-            .cast<BleCharacteristic?>()
-            .firstWhere(
-                (char) =>
-                    char != null && char.id == viamManagedMachineNameCharUUID,
-                orElse: () => null);
+        BleCharacteristic? periphNameChar;
+        for (int i = 0; i < numRetries; i++) {
+          periphNameChar = viamSvc.characteristics
+              .cast<BleCharacteristic?>()
+              .firstWhere(
+                  (char) =>
+                      char != null && char.id == viamManagedMachineNameCharUUID,
+                  orElse: () => null);
+          if (periphNameChar != null) {
+            break;
+          }
+        }
         if (periphNameChar == null) {
-          // Note(erd): this could use some retry logic
-          print(
+          logger.e(
               'did not find needed periph name char after discovery; disconnecting');
           await periph.disconnect();
           deviceSub.resume();
           return;
         }
 
         final periphName = utf8.decode((await periphNameChar.read())!);
-        if (periphName != machineName) {
-          // Note(erd): this could use some retry logic
```

Comment:
Did _not_ add retry logic to this one (reading and decoding the char value). Once the characteristic has been discovered, is there really a case where the value might not yet be readable?